### DISCUSSION TITLE
Multi-level localization fallback

### DISCUSSION THREAD

**Discussion by marklonquist** [payloadcms/payload#2933] [Feature Requests & Ideas]
Hi

Having a deeper fallback structure for localization can in some cases make the job of a content editor a lot easier.

Structure example:

- en
- de
  - fallback: en
- de-de
  - fallback: de
- de-at
  - fallback: de

so, in this case, if the value is not filled on `de-de`, we take the value from `de`.
If that's not filled, we fallback to `en` which - hopefully - is present.

**Comment by DanRibbens** [comment]
I like this idea and it would fit nicely into the config where we have extended the locales to be objects instead of an array of strings to support additional properties. That work is on the [feat/rtl-support](https://github.com/payloadcms/payload/tree/feat/rtl-support) branch.

After that gets merged, we could add `fallbackLocale` and extend the fallback logic to work perfectly.

**Comment by marklonquist** [comment]
Any update on this feature? We can't really change to using payload before we have this :)

**Reply by DanRibbens** [reply]
The LTR PR is slated for our 2.0 changes which is complicating things a little as we work towards a larger release. I can't give an exact timeline, but you should know we are in the weeds with some very exciting changes coming!

**Reply by marklonquist** [reply]
Thank you for the update :)

**Reply by marklonquist** [reply]
We are not in a rush to change. Our current platform works, but payload is much faster :)

**Comment by marklonquist** [comment]
having installed 2.0 now, I can see this is still not available 😭

**Reply by marklonquist** [reply]
@DanRibbens any info to share? :)

**Reply by DanRibbens** [reply]
It didn't make it into 2.0 with all the other changes we had to prioritize. This is still high on my list after  we clean up our issues.

**Reply by marklonquist** [reply]
Cool. Thanks for the update. 

It’s a vital part on how we use a CMS so we need it before we can make the transition to payload. :-)

So really looking forward to this feature.

**Reply by marklonquist** [reply]
@DanRibbens any progress on this? :)

**Comment by marklonquist** [comment]
It seems this was PARTIALLY implemented.

It's now possible to set a localized fallback, but if that fallback does not have any data, it doesn't seem to use the fallbacks fallback langauage, eg.:

- en
- de
    - fallback: en
- de-de
    - fallback: de
- de-at
    - fallback: de
    
    
Let's say we have a tittle field. On `de-de` it's not set, so it fallbacks to `de`, but on `de` it's also not set, so the value currently returns as empty. But I would think it should return the value of `en`, as `de` fallbacks to that language.

**Reply by marklonquist** [reply]
@DanRibbens sorry to ping you again, but is this in your plans to make? :)

**Reply by DanRibbens** [reply]
Your understanding is correct on the current way fallbacks are working. I am unsure if we will be able to achieve fallbacks more than one extra language. 

I would also be concerned with introducing circular dependency issues if we made this any more complex.

**Reply by marklonquist** [reply]
> Your understanding is correct on the current way fallbacks are working. I am unsure if we will be able to achieve fallbacks more than one extra language. 
> 
> I would also be concerned with introducing circular dependency issues if we made this any more complex.
> 

That sounds weird though, as other CMS' support this. 

Can you point me in the right direction if I were to make this change myself?

**Comment by marklonquist** [comment]
Nevermind - our ecom team do not need more than what is already provided. Closing this and thanks! 😄
